core/src/nodes/raw_swarm.rs

```diff
@@ -1371,7 +1371,7 @@ where
 {
     /// If we are connected, returns the `PeerConnected`.
     #[inline]
-    pub fn as_connected(self) -> Option<PeerConnected<'a, TInEvent, TPeerId>> {
+    pub fn into_connected(self) -> Option<PeerConnected<'a, TInEvent, TPeerId>> {
```

Comment:
Not a fan of turning `as` into `into`, as we return an `Option`.

Reply:
https://rust-lang.github.io/rust-clippy/master/index.html#wrong_self_convention

The prefixes are more concerned with the use of `self` and since it is consumed here the `into` seems appropriate to me despite the `Option`.

Reply:
I think that my original thought was that `Peer` is kind of a pointer. It is borrowing from the `RawSwarm`. It doesn't implement `Copy` because it is a mutable borrow.
Therefore the prefix is `as` because we're turning "a pointer" into "another pointer", and in the collective imagination converting a pointer would in general be `as` and not `into`.